src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -572,6 +572,13 @@ private void ProcessEvents()
                             expandedName = associatedDirectoryEntry.GetPath(true, nextEvent.name);
                         }
 
+                        // To match Windows, ignore all changes that happen on the root folder itself
+                        if (expandedName == string.Empty)
```

Comment:
Hmm good point; @stephentoub  would this be better with a string.IsNullOrEmpty check?